### PR TITLE
Domicile transfer/pdf creator

### PR DESCRIPTION
**Pull request by Rikkiabb**
# \<Description\>
Endpoint for creating pdf for children legal residence transfer. The process is the following:

Create pdf from application.
Upload pdf to s3 bucket.
Create presigned url for bucket.
Return presigned url to be used in client.
The endpoint takes in an application and a type which refers to a pdf type. Current types for pdf's is only one: children legal residence transfer, but when more pdf's are needed then we can add to this type and create a template for it and that's it.

The endpoint uses pdfkit. I searched slack and saw talk about IText Pdf for pdf creation (https://islandis.slack.com/archives/C019VN1KG04/p1599483604046900?thread_ts=1599468993.039200&cid=C019VN1KG04) but didn't find any usage for it in the monorepo but found that pdfkit was used so went with that for now. This decision can and in my opinion should be revisited on a larger scale than this PR.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by darrikonn — APPROVED** [review]
core/devops files look good

File: apps/application-system/api/src/environments/environment.ts [hunk 1]
```diff
@@ -25,4 +25,5 @@ export default {
   auth: {
     jwtSecret: 'supersecret',
   },
+  fsS3Bucket: process.env.FILE_SERVICE_PRESIGN_BUCKET,
```
Comment (Rikkiabb) [inline apps/application-system/api/src/environments/environment.ts:28]: Is this always read from the proccess environment or is sometimes hardcoded for local development?

File: apps/application-system/api/src/app/modules/application/files/file.service.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      .upload(uploadParams)
+      .promise()
+      .catch(() => {
+        return null
+      })
+
+    const presignedUrlParams = {
+      Bucket: bucket,
+      Key: fileName,
+      Expires: this.one_minute * 120, // TODO: Select length for presigned url's in island.is
```
Comment (Rikkiabb) [inline apps/application-system/api/src/app/modules/application/files/file.service.ts:117]: Does island.is have any guidelines on how long presigned urls should be active? Currently this is just a random number....

File: apps/application-system/api/src/app/modules/application/files/file.service.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    const parentA: ParentResidenceChange = {
+      id: nationalRegistryData.nationalId,
+      name: nationalRegistryData.fullName,
+      ssn: nationalRegistryData.nationalId,
+      phoneNumber: answers.phoneNumber as string,
+      email: answers.email as string,
+      address: nationalRegistryData.address?.streetAddress as string,
+      postalCode: nationalRegistryData.address?.postalCode as string,
+      city: nationalRegistryData.address?.city as string,
+    }
```
Comment (Rikkiabb) [inline apps/application-system/api/src/app/modules/application/files/file.service.ts:67]: I added tests for this. But is there a way to get type safety for this? Or do we always have to this thingamajig and trust the type will be cast to the correct type?

File: libs/application/api-template-utils/src/lib/types.ts [hunk 4]
```diff
@@ -19,3 +19,7 @@ export type SendEmail = { type: 'sendEmail'; template: SendMailOptions }
 export type ApplicationAPITemplateAction =
   | SendEmail
   | AssignApplicationThroughEmail
+
+export enum PDF_TYPES {
```
Comment (jeremybarbet) [inline libs/application/api-template-utils/src/lib/types.ts:23]: A bit weird to have this type located here, while the api-template-utils doesn't do anything regarding PDF. I would define it locally in your app instead and if it's need to be shared between libraries and app then better to define inside shared/tyoes

**Review by VilliGunn — APPROVED** [review]
No further comments from me since the first review. Nice changes 💯

File: libs/application/api-template-utils/src/lib/types.ts (hunk 4 above)
Comment (Rikkiabb) [reply libs/application/api-template-utils/src/lib/types.ts:23]: Totally agree, will move 👍

**Review by jeremybarbet — APPROVED** [review]
LGTM 👍

**Review by darrikonn — APPROVED** [review]
devops/core files look good

**Review by atligudl — APPROVED** [review]

**Review by pshomov — APPROVED** [review]
LGTM

**Review by VilliGunn — APPROVED** [review]

**Review by Thorrik58 — APPROVED** [review]

**Pull request merged**